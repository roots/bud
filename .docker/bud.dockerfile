FROM node:lts

LABEL name 'bud'
LABEL version 1

RUN npm install -g verdaccio pm2

COPY ./.docker/bud/motd /etc/motd
COPY ./.docker/bud/bash.bashrc /etc/bash.bashrc
