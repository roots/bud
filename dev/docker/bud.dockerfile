FROM node:16

LABEL name 'bud'
LABEL version 1

RUN npm install netlify-cli npm-cli-login --global

COPY ./dev/docker/bud/motd /etc/motd
COPY ./dev/docker/bud/bash.bashrc /etc/bash.bashrc

COPY ./ /srv/bud

RUN mkdir -p /srv/bud/node_modules \
  && mkdir -p /srv/bud/storage/verdaccio \
  && mkdir -p /srv/bud/storage/yarn \
  && mkdir -p /srv/bud/storage/coverage \
  && mkdir -p /srv/mocks
