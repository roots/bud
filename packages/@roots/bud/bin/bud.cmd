@echo off

node "%~dp0\bud.js" %*
