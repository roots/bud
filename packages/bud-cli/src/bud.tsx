#!/usr/bin/env node
import {yargs} from '@roots/bud-support'
import * as build from './commands/build'
import * as publish from './commands/publish'

/**
 * $bud
 */
const cli = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(1, 'Try using one of the above commands.')
  .usage('bud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .showHelpOnFail(true)
  .epilog('https://github.com/roots/bud').argv

export {cli as bud}
module.exports = {bud: cli}
