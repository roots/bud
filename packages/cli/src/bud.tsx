#!/usr/bin/env node

import {yargs} from '@roots/bud-support'
import * as build from './commands/build'
import * as publish from './commands/publish'

module.exports = yargs
  .command(build)
  .command(publish)
  .recommendCommands()
  .demandCommand(1, 'Try using one of the above commands.')
  .usage('bud [command] [options]')
  .version()
  .wrap(yargs.terminalWidth())
  .showHelpOnFail(true)
  .epilog('https://github.com/roots/bud').argv
