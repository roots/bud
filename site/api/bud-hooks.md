---
id: bud-hooks
title: bud-hooks package
sidebar_label: bud-hooks package
hide_title: true
sidebar: "api"
slug: bud-hooks
---

<!-- Do not edit this file. It is automatically generated by API Documenter. -->

## bud-hooks package

Hooks system used for framework eventing.

## Remarks

- 💁 Composable - Build exceptional applications with a modular, configurable build system

- 💪 Modern - Modern framework that scales from a single file to thousands of lines of code

- 🌱 Easy - Low bundle size and fast build times

## Classes

| Class                         | Description                                             |
| ----------------------------- | ------------------------------------------------------- |
| [Hooks](/api/bud-hooks/hooks) | Service allowing for fitering values through callbacks. |
