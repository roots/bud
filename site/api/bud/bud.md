---
id: bud.bud
title: bud namespace
sidebar_label: bud namespace
hide_title: true
sidebar: "api"
slug: bud
---

<!-- Do not edit this file. It is automatically generated by API Documenter. -->

## Bud namespace

Signature:

```typescript
export declare namespace Bud
```

## Type Aliases

| Type Alias                      | Description             |
| ------------------------------- | ----------------------- |
| [Options](/api/bud/bud/options) | Bud constructor options |
