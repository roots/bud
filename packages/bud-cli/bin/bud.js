#!/usr/bin/node
require('./../lib/bud')
