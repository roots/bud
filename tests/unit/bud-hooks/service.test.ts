import {beforeAll, describe, expect, it, jest} from '@jest/globals'
import {Bud, factory} from '@repo/test-kit/bud'
import {Hooks} from '@roots/bud-hooks'

describe('@roots/bud-hooks', function () {
  let bud: Bud

  let hooks: Hooks

  beforeAll(async () => {
    bud = await factory()
    hooks = new Hooks(bud)
    hooks.store = {} as any
  })

  it('has an on method', () => {
    expect(hooks.on).toBeInstanceOf(Function)
  })

  it('has an async method', () => {
    expect(hooks.on).toBeInstanceOf(Function)
  })

  it('has a filter method', () => {
    expect(hooks.filter).toBeInstanceOf(Function)
  })

  it('has a filterAsync method', () => {
    expect(hooks.on).toBeInstanceOf(Function)
  })

  it('has an action method', () => {
    expect(hooks.action).toBeInstanceOf(Function)
  })

  it('has a fire method', () => {
    expect(hooks.fire).toBeInstanceOf(Function)
  })

  it('async registers value', () => {
    const callback = async () => 'bar'
    // @ts-ignore
    hooks.async('build.resolve', callback)
    expect(hooks.store['build.resolve']).toStrictEqual([callback])
  })

  it('filterAsync retrieves value', async () => {
    const value = await hooks.filterAsync('build.resolve')
    expect(value).toBe('bar')
  })

  it('action registers callable function', async () => {
    const value = jest.fn(async (app: Bud) => null)
    // @ts-ignore
    hooks.action('event.app.build', value)
    expect(hooks.store['event.app.build'].pop()).toBe(value)
  })

  it('fire calls action function', async () => {
    const value = jest.fn(async () => null)
    hooks.action('event.build.before', value)
    await hooks.fire('event.build.before')
    expect(value).toHaveBeenCalled()
  })
})
