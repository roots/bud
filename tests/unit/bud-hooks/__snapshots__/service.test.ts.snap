// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@roots/bud-hooks hooks repository matches snapshot 1`] = `
Object {
  "build": Array [
    [Function],
  ],
}
`;
